frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -1,89 +1,110 @@
 import * as React from 'react';
-import { useFormikContext, FormikValues, useField } from 'formik';
+import { useFormikContext, FormikValues, FormikTouched } from 'formik';
 import { Alert, TextInputTypes, ValidatedOptions } from '@patternfly/react-core';
 import { getGitService, GitProvider } from '@console/git-service';
-import { InputField, DropdownField, useFormikValidationFix } from '@console/shared';
+import {
+  InputField,
+  DropdownField,
+  useFormikValidationFix,
+  useDebounceCallback,
+} from '@console/shared';
 import { GitReadableTypes, GitTypes } from '../import-types';
 import { detectGitType, detectGitRepoName } from '../import-validation-utils';
 import { getSampleRepo, getSampleRef, getSampleContextDir } from '../../../utils/imagestream-utils';
 import FormSection from '../section/FormSection';
 import SampleRepo from './SampleRepo';
 import AdvancedGitOptions from './AdvancedGitOptions';
-import { UNASSIGNED_KEY } from '../../../const';
+import { UNASSIGNED_KEY, CREATE_APPLICATION_KEY } from '../../../const';
 
 export interface GitSectionProps {
   showSample?: boolean;
 }
 
 const GitSection: React.FC<GitSectionProps> = ({ showSample }) => {
-  const { values, setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
-  const [, { touched: gitTypeTouched }] = useField('git.type');
-  const [, { touched: userSelection }] = useField('image.selected');
+  const { values, setFieldValue, setFieldTouched, touched, dirty } = useFormikContext<
+    FormikValues
+  >();
   const tag = values.image.tagObj;
   const sampleRepo = showSample && getSampleRepo(tag);
+  const { application = {}, name: nameTouched, git = {} } = touched;
+  const { type: gitTypeTouched } = git as FormikTouched<{
+    type: boolean;
+    url: boolean;
+    urlValidation: boolean;
+  }>;
+  const { name: applicationNameTouched } = application as FormikTouched<{ name: boolean }>;
+  const [validated, setValidated] = React.useState<ValidatedOptions>(ValidatedOptions.default);
 
-  const handleGitUrlBlur: React.ReactEventHandler = React.useCallback(async () => {
-    const { url, ref } = values.git;
+  const handleGitUrlChange = React.useCallback(
+    async (url: string, ref: string) => {
+      setFieldValue('git.isUrlValidating', true);
+      setValidated(ValidatedOptions.default);
 
-    setFieldValue('git.isUrlValidating', true);
+      const gitType = gitTypeTouched ? values.git.type : detectGitType(url);
+      const gitRepoName = detectGitRepoName(url);
+      const showGitType = gitType === GitTypes.unsure || gitTypeTouched;
 
-    const gitType = detectGitType(url);
-    const gitRepoName = detectGitRepoName(url);
-    const showGitType = gitType === GitTypes.unsure;
+      setFieldValue('git.type', gitType);
+      setFieldValue('git.showGitType', showGitType);
+      showGitType && setFieldTouched('git.type', false);
```

Comment:
Setting the `git.type` touched flag to false means that a manually selected type jumps back to other if the type was not successfully selected. If I understand the diff correctly this happens also in the old version. If is possible to reset the touched only if the git type auto detection was successful? Or skip this in any case?

I comment out this line in my local code which results in an endless loop of GitLab API requests. So it more complex than just skip this line.

Testing:
1. Set URL to "https://version.helsinki.fi/tommi.tuura/public-project"
2. Select GitLab
3. Add ".git" to the end of the URL